binder/src/main/java/io/grpc/binder/SecurityPolicies.java

```diff
@@ -191,6 +194,12 @@ private static boolean checkPackageSignature(
    * Creates a {@link SecurityPolicy} that allows access if and only if *all* of the specified
    * {@code securityPolicies} allow access.
    *
+   * <p>Policies will be checked in the order that they are passed. If a policy denies access,
```

Comment:
I thought about adding this to the contract too but wasn't sure since it would preclude ever (optimistically) checking the policies in parallel. Is there a specific need for or benefit of this guarantee that you have in mind?

Reply:
No. And I'm happy not committing to a particular behavior, but in that case I would prefer to be explicit that we intentionally make no guarantees. Something like "subsequent policies may be checked in any order and on any thread" so that we are clear you should not depend on that behavior.

Reply:
Got it thanks. Thinking about this more with http://www.hyrumslaw.com in mind, I doubt anyone would ever be brave enough to make such a change even if we added an explicit non-guarantee to this javadoc. So maybe your proposal to guarantee the order is for the best since it is the simplest. Should be a separate PR though?

Reply:
Removed all allOf() changes from this PR.